tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/LoginSteps.cs

```diff
@@ -30,10 +40,16 @@ public void WhenALoginRequestIsMade(string username, string password)
             _seleniumContext.WebDriver.FindElement(By.TagName("form")).Submit();
         }
 
-        [Then(@"The user is redirected to (.*)")]
-        public void ThenTheUserIsRedirectedTo(string url)
+        [Then(@"the user is redirected to (identity server|client) page (.*)")]
+        public void ThenTheUserIsRedirectedTo(string target, string url)
+        {
+            _seleniumContext.WebWaiter.Until(x => x.Url.StartsWith($"{HostUrls[target]}{url}",StringComparison.OrdinalIgnoreCase));
+        }
+
+        [Then(@"the page contains paragraph with text (.*)")]
+        public void ThenThePageVerifiesItCouldTalkToTheSampleResource(string text)
         {
-            _seleniumContext.WebWaiter.Until(x => x.Url == url);
+            _seleniumContext.WebDriver.FindElement(By.Id("sampleResourceResult")).Text.Should().Be(text);
```

Comment:
We had a quick discussion on this. I thought we were going to try keep this a bit more generic? By making it generic to find a particular tag or element we will introduce a set of common steps that other tests will be able to use.

Reply:
Sure, we can do that